Add paintings and fix issues with Museum Room

taesungh opened the pull request:
Add paintings to Carousel Gallery
- Insert paintings provided by graphics into Carousel Gallery
- Add aria-label with description of artwork
- Double rotation period when prefers-reduced-motion

Fix perspective origin issues on Safari 15 and iOS
- The perspective origin would be incorrectly calculated, causing the back side of the carousel to appear in view
- Remove the inherited perspective style on the gallery and instead preserve the 3D transform for the parent to apply perspective

Restore desired Museum Room ceiling overflow
- Include the Museum Room ceiling out of view so the ceiling grows when expanding a collapsed navigation bar

Fix sass bundling issue with `radial-gradient`
- The first color stop is forced to 0% in the outputted CSS, so manually include a color for the 0% stop

taesungh committed:
Add paintings to Carousel Gallery

- Insert paintings provided by graphics into Carousel Gallery
- Add aria-label with description of artwork
- Double rotation period when prefers-reduced-motion

taesungh committed:
Fix perspective origin issues on Safari 15 and iOS

- The perspective origin would be incorrectly calculated, causing the
  back side of the carousel to appear in view
- Remove the inherited perspective style on the gallery and instead
  preserve the 3D transform for the parent to apply perspective

taesungh committed:
Restore desired Museum Room ceiling overflow

- Include the Museum Room ceiling out of view so the ceiling grows when
  expanding a collapsed navigation bar

taesungh committed:
Fix sass bundling issue with `radial-gradient`

- The first color stop is forced to 0% in the outputed CSS, so manually
  include a color for the 0% stop

samderanova commented on src/components/CarouselGallery/CarouselGallery.module.scss:
```diff
@@ -32,16 +32,18 @@
-			animation: $tt rotate-carousel infinite linear;
+			animation: rotate-carousel infinite linear;
+			animation-duration: $rotation-period;
+			@media (prefers-reduced-motion: reduce) {
+				animation-duration: 2 * $rotation-period;
+			}
```
I was under the impression that for users who prefer reduced motion, we would just not rotate the carousel at all, but this is fine, especially since it's probably easier this way.

taesungh replied:
I suppose it could go either way, not sure what would be best.

RandyHuynh5815 commented on src/components/CarouselGallery/CarouselGallery.tsx:
```diff
@@ -1,23 +1,47 @@
 }
 
+const PAINTINGS = [
+	{ image: mona_petr, aspectRatio: "4/5" },
```
Question: How is aspectRatio used? It is not passed through the mapping, and it is not used in the CSS files.

taesungh replied:
Currently it is not. I originally included it to preserve the original aspect ratio of the paintings rather than cropping, but that would result in awkward spacing between paintings since they are equally spaced radially, so we would need to manually adjust offset angles. We may revisit this in the future.

RandyHuynh5815 reviewed (commented):
Thank you for fixing this. I just have a small question, but besides that, everything should be good. 

Also for the landing page issue, the wall panel background image attempts to maintain its aspect ratio, so its height as changes as we change the width. As a result, there is a breakpoint where the wall panel would not be able to properly reach the ground from the ceiling without covering the entire room. Is it possible to shrink the height of the museum room along with the width at a certain breakpoint? We can discuss more if needed.

taesungh committed:
Fix gallery painting sizing

- Add `sizes` attribute to optimize resolution
- Change `object-fit` to cover instead of stretch

samderanova commented on src/components/CarouselGallery/CarouselGallery.tsx:
```diff
@@ -39,7 +39,15 @@ function CarouselGallery({ rotating }: CarouselGalleryProps) {
-						<Image src={image} alt="" fill priority placeholder="blur" />
+						<Image
+							src={image}
+							alt=""
```
Is it possible to provide alt text for these images?

RandyHuynh5815 replied:
I think it can be done if `name` key to the objects in `PAINTINGS` and pass those as `alt`.

taesungh replied:
The images on the carousel are repeated and moving, so we consider them to be decorative, but the entire carousel has an `aria-label` description.

samderanova replied:
Alright sounds good
